mountpoint-s3/src/main.rs

```diff
@@ -429,24 +429,27 @@ fn create_client_for_bucket(
     mut endpoint_config: EndpointConfig,
     client_config: S3ClientConfig,
 ) -> Result<S3CrtClient, anyhow::Error> {
-    let region_to_try = supposed_region.clone().unwrap_or_else(|| {
+    const DEFAULT_REGION: &str = "us-east-1";
+
+    let region_to_try = supposed_region.as_deref().unwrap_or_else(|| {
         if endpoint_url.is_some() {
             tracing::warn!(
                 "endpoint specified but region unspecified. using {} as the signing region.",
-                endpoint_config.get_region()
+                DEFAULT_REGION
             );
         }
-        endpoint_config.get_region().to_owned()
+        DEFAULT_REGION
     });
+    endpoint_config = endpoint_config.region(region_to_try);
```

Comment:
Right. I was only once setting the region that was default and not after getting the `region_to_try` in endpoint config. Which meant it was always `us-east-1` in `EndpointConfig`. Thanks.